packages/js/onboarding/src/components/WooPaymentsMethodsLogos/index.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect } from 'react';
+import { Fragment, createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Visa from '../../images/cards/visa';
+import MasterCard from '../../images/cards/mastercard';
+import Amex from '../../images/cards/amex';
+import Discover from '../../images/cards/discover';
+import ApplePay from '../../images/cards/applepay';
+import GooglePay from '../../images/cards/googlepay';
+import JCB from '../../images/cards/jcb';
+import WooPay from '../../images/payment-methods/woopay';
+import AfterPay from '../../images/payment-methods/afterpay';
+import Affirm from '../../images/payment-methods/affirm';
+import Klarna from '../../images/payment-methods/klarna';
+
+/**
+ * Payment methods logos
+ */
+const PaymentMethods = [
+	{
+		name: 'visa',
+		component: <Visa key="visa" />,
+	},
+	{
+		name: 'mastercard',
+		component: <MasterCard key="mastercard" />,
+	},
+	{
+		name: 'amex',
+		component: <Amex key="amex" />,
+	},
+	{
+		name: 'discover',
+		component: <Discover key="discover" />,
+	},
+	{
+		name: 'woopay',
+		component: <WooPay key="woopay" />,
+	},
+	{
+		name: 'applepay',
+		component: <ApplePay key="applepay" />,
+	},
+	{
+		name: 'googlepay',
+		component: <GooglePay key="googlepay" />,
+	},
+	{
+		name: 'afterpay',
+		component: <AfterPay key="afterpay" />,
+	},
+	{
+		name: 'affirm',
+		component: <Affirm key="affirm" />,
+	},
+	{
+		name: 'klarna',
+		component: <Klarna key="klarna" />,
+	},
+	{
+		name: 'jcb',
+		component: <JCB key="jcb" />,
+	},
+];
+
+// Maximum number of logos to be displayed on a mobile screen.
+const maxElementsMobile = 5;
+// Maximum number of logos to be displayed on a tablet screen.
+const maxElementsTablet = 7;
+// Maximum number of logos to be displayed on a desktop screen.
+const maxElementsDesktop = 10;
+// Total number of available payment methods.
+const totalPaymentMethods = 21;
+
+export const WooPaymentMethodsLogos: React.VFC< {
+	isWooPayEligible: boolean;
+	maxElements: number;
+} > = ( { isWooPayEligible = false, maxElements = maxElementsDesktop } ) => {
+	const [ maxShownElements, setMaxShownElements ] = useState( maxElements );
+
+	// Determine the maximum number of logos to display, taking into account WooPay’s eligibility.
+	const getMaxShownElements = (
+		maxElementsNumber: number,
+		isWooPayAvailable: boolean
+	) => {
+		if ( ! isWooPayAvailable ) {
+			return maxElementsNumber + 1;
```

Comment:
Shouldn't this be minus 1?

Reply:
Not really. The purpose of +1 is to exclude WooPay if it is not eligible, resulting in fewer logos being displayed. That’s why we increase the maxElements by one.